### PR TITLE
Add tests to builders

### PR DESCRIPTION
**Pull request by franmomu**

### CODE REVIEW

File: tests/Builder/DatagridBuilderTest.php [hunk 1]
```diff
@@ -111,10 +111,48 @@ public function testGetBaseDatagrid(): void
...
     public function testFixFieldDescription(): void
     {
+        $fieldMapping = [
+            'fieldName' => 'plainField',
+            'name' => 'plainField',
+            'columnName' => 'plainField',
+            'type' => 'string',
+        ];
+
         $classMetadata = $this->createMock(ClassMetadata::class);
```
Comment (phansys) [inline tests/Builder/DatagridBuilderTest.php:121]:
```suggestion
        $classMetadata = $this->createStub(ClassMetadata::class);
```

File: tests/Builder/DatagridBuilderTest.php [hunk 2]
```diff
@@ -111,10 +111,48 @@ public function testGetBaseDatagrid(): void
...
+        $this->assertTrue($fieldDescription->getOption('global_search'));
+    }
+
+    public function testFixFieldDescriptionWithAssociationMapping(): void
+    {
+        $associationMapping = [
+            'fieldName' => 'associatedDocument',
+            'name' => 'associatedDocument',
+        ];
+        $classMetadata = $this->createMock(ClassMetadata::class);
```
Comment (phansys) [inline tests/Builder/DatagridBuilderTest.php:148]:
```suggestion
        $classMetadata = $this->createStub(ClassMetadata::class);
```

File: tests/Builder/FormContractorTest.php [hunk 3]
```diff
@@ -145,4 +145,53 @@ public function testAdminClassAttachForNotMappedField(): void
         // When
         $this->formContractor->fixFieldDescription($admin, $fieldDescription);
     }
+
+    public function testFixFieldDescriptionForFieldMapping(): void
+    {
+        $fieldMappings = [
+            'type' => 'string',
+        ];
+        $classMetadata = $this->createMock(ClassMetadata::class);
```
Comment (phansys) [inline tests/Builder/FormContractorTest.php:154]:
```suggestion
        $classMetadata = $this->createStub(ClassMetadata::class);
```

File: tests/Builder/FormContractorTest.php [hunk 4]
```diff
@@ -145,4 +145,53 @@ public function testAdminClassAttachForNotMappedField(): void
...
+        $this->assertSame($fieldMappings, $fieldDescription->getFieldMapping());
+    }
+
+    public function testFixFieldDescriptionForAssociationMapping(): void
+    {
+        $associationMappings = [
+            'fieldName' => 'test',
+            'type' => 'string',
+        ];
+        $classMetadata = $this->createMock(ClassMetadata::class);
```
Comment (phansys) [inline tests/Builder/FormContractorTest.php:179]:
```suggestion
        $classMetadata = $this->createStub(ClassMetadata::class);
```

File: tests/Fixtures/Document/SimpleAnnotationDocument.php [hunk 5]
```diff
@@ -24,6 +24,19 @@ class SimpleAnnotationDocument
     /** @ODM\Field(type="string") */
     private $name;
 
+    /**
+     * @ODM\EmbedOne()
+     */
+    private $associatedDocument;
+
+    /** @ODM\EmbedMany() */
```
Comment (phansys) [inline tests/Fixtures/Document/SimpleAnnotationDocument.php:32]:
```suggestion
    /**
     * @ODM\EmbedMany()
     */
```

File: tests/Builder/DatagridBuilderTest.php (hunk 1 above)
Comment (franmomu) [reply tests/Builder/DatagridBuilderTest.php]: Thanks! I ended up using actual `ClassMetadata` instances.

**Comment by jordisala1991** [comment]
Thank you @franmomu

**Pull request merged**